.env.template

```diff
@@ -1,4 +1,47 @@
-MONGO_INITDB_ROOT_USERNAME=
-MONGO_INITDB_ROOT_PASSWORD=
-MONGO_DB_USER=
-MONGO_DB_PASSWORD=
\ No newline at end of file
+## GENERAL ##
+# Running through Docker, use: NODE_ENV=production
+NODE_ENV=production
+SERVER=localhost
+
+## DATABASE ##
+# Running through Docker: MONGO_HOST=mongodb
+# Running outside Docker: MONGO_HOST=localhost
+MONGO_HOST=mongodb
+MONGO_DB=praise_db 
+MONGO_PORT=27017
+MONGO_INITDB_ROOT_USERNAME=praiseDbRootUsername
+MONGO_INITDB_ROOT_PASSWORD=[any password]
+MONGO_USERNAME=praiseDbUsername
+MONGO_PASSWORD=[any password]
+
+## API ##
+API_PORT=8099
+ADMINS=0x123..123,0x123..12
+
+## API AUTHENTICATION ##
+COOKIE_DOMAIN=localhost
+COOKIE_PATH=/
+SECURE_COOKIE=false
+JWT_SECRET=xxxxxxxxxxxxxx
+COOKIE_SECRET=xxxxxxxxxxxxxx
+# expires after 1 hour of inactivity, or 3 days
+JWT_ACCESS_EXP=3600
+JWT_REFRESH_EXP=25920000
+
+## FRONTEND ## 
+# Running through Docker, leave empty: REACT_APP_API_URL=
+# Running outside Docker, set to SERVER:API_PORT: REACT_APP_API_URL=http://localhost:8088
+REACT_APP_API_URL=
+FRONTEND_DEVELOPMENT_PORT=3000
+
+## DISCORD_BOT ##
+DISCORD_TOKEN=
+DISCORD_CLIENT_ID=
+DISCORD_GUILD_ID=
+
+## LOGGING ##
+JET_LOGGER_MODE=CONSOLE
+JET_LOGGER_FILEPATH=jet-logger.log
+JET_LOGGER_TIMESTAMP=TRUE
+JET_LOGGER_FORMAT=LINE
+
```

Comment:
I don't think all the env variables for 3 applications should be kept in a single file. What if we want some variables to only be available to a single application and not others (like secret keys)?

Only the shared env variables should be in the shared .env files. Application-specific env variables should be within their own workspace. I think that was the correct approach.

Reply:
You can specify multiple env files for each docker application. See https://github.com/docker/compose/issues/6392#issuecomment-444265127

Reply:
Yes, not as straight forward though specifying more than one env file outside of Docker. One alternative is to let packages load env as before, except when running inside Docker. 

How to know when running inside Docker? There is a library for that: https://www.npmjs.com/package/is-docker

Reply:
> The env variables for 3 applications should not be kept in a single file. What if we want some variables to only be available to a single application and not others (like secret keys)?
> 
> Only the shared env variables should be in the shared .env files. Application-specific env variables should be within their own workspace. I think that was the "best practice" approach.

Yeah, I intentionally disregarded the logger settings. Jet logger producing log files inside the Docker container is not ideal anyway, a revisit to the logging setup is needed. Let's make a separate issue for that one.

Reply:
> Yeah, I intentionally disregarded the logger settings. Jet logger producing log files inside the Docker container is not ideal anyway, a revisit to the logging setup is needed. Let's make a separate issue for that one.

Hmm -- you could setup a docker volume to mount a path in system filesystem with the docker container's filesystem. See https://docs.docker.com/storage/volumes/#use-a-volume-with-docker-compose

Reply:
I believe the output from jet-logger made to stdout is caught by Docker and stored outside the container. Let's investigate that before looking into other solutions.